Architecture/TodoApp/android/core/src/main/java/com/novoda/todoapp/tasks/service/PersistedTasksService.java

```diff
@@ -335,4 +356,22 @@ public boolean ifMatches(SyncedData<Task> value) {
         };
     }
 
+    private static Observable.Transformer<Tasks, Event<Tasks>> asValidatedEvent(final Event<Tasks> value) {
+        return new Observable.Transformer<Tasks, Event<Tasks>>() {
+            @Override
+            public Observable<Event<Tasks>> call(Observable<Tasks> tasksObservable) {
+                return tasksObservable.compose(EventFunctions.asEvent(value)).map(new Func1<Event<Tasks>, Event<Tasks>>() {
```

Comment:
done in https://github.com/novoda/spikes/pull/113/commits/fe9a9d11ad7afecd4bfa7ad5d03e5a15dc5575b1